external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+if(${ENABLE_gauxc})
+    if((NOT "${gauxc_DIR}" STREQUAL "") AND "${IntegratorXX_DIR}" STREQUAL "")
+       set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+    endif()
+
+    find_package(ExchCXX CONFIG REQUIRED PATHS ${gauxc_DIR}/../ExchCXX) 
+    find_package(gauxc CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== check if GPU-enabled GauXC is found, if GPU is specified ==#
+        get_property(_gpu TARGET gauxc::gauxc PROPERTY GAUXC_HAS_DEVICE)
+        
+        if(${gauxc_ENABLE_GPU} AND (NOT ${_gpu}))
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned on, but selected GauXC install is not built for GPU support!")
+        elseif((NOT ${gauxc_ENABLE_GPU}) AND ${_gpu})
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned off, but selected GauXC install is built for GPU support!")
+        endif()
+
+        #== normal print-out and library addition ==#
+        get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION_RELWITHDEBINFO)
+        #get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION)
+        message(STATUS "${Cyan}Found gauxc ${ColourReset}: ${_loc} (found version ${gauxc_VERSION})")
+        add_library(gauxc_external INTERFACE)  # dummy
+
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(VERBOSE "Suitable gauxc located externally as user insists")
+        endif()
+
+    else()
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(FATAL_ERROR "Suitable gauxc could not be externally located as user insists")
+        endif()
+
+        message(STATUS "Suitable gauxc could not be located, ${Magenta}Building gauxc${ColourReset} instead.")
+
+        include(ExternalProject)
+        ExternalProject_Add(gauxc_external
+            #URL https://github.com/wavefunction91/GauXC/archive/issue_56_58/reduce_shell_pair_mem.tar.gz 
+            URL https://github.com/wavefunction91/GauXC/archive/7746c2c.tar.gz 
+            UPDATE_COMMAND ""
+            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
+                   -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                   -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                   -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                   -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
+                   -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
+                   -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
+                   -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT}
+                   -DGAUXC_ENABLE_CUDA=${gauxc_ENABLE_GPU}
```

Comment:
Other review comments bring up an auxiliary question here. Internal construction of Gauxc within Psi4 feeds the `gauxc_ENABLE_GPU` Psi4 keyword in to the `GAUXC_ENABLE_CUDA` keyword of GauXC; so currently, only CUDA is supported for internally-constructed GauXC instances within Psi4. Is this acceptable? Or should we allow HIP support for internally-built GauXC instances? Presumably if we allow HIP support, we would want to split up `gauxc_ENABLE_GPU` into `gauxc_ENABLE_CUDA` and `gauxc_ENABLE_HIP` keywords Psi4-side.

Reply:
Yes, that's what you'd want to do if you want to support the full feature set. But like I said (elsewhere, can't dig up the comment for some reason), the HIP version is lagging a bit. might make sense to just hold off on that and have GPU == CUDA for the time being

Reply:
That makes sense. I'm happy with limiting internal build support to just CUDA for now then, at least until the HIP implementation is deemed more up-to-date.